tests/Nox.Types.Tests/NoxCountryCode3Test.cs

```diff
@@ -0,0 +1,299 @@
+    [InlineData("MWI")]
+    [InlineData("MYS")]
+    [InlineData("MDV")]
+    [InlineData("MLI")]
+    [InlineData("MLT")]
+    [InlineData("MHL")]
+    [InlineData("MTQ")]
+    [InlineData("MRT")]
+    [InlineData("MUS")]
+    [InlineData("MYT")]
+    [InlineData("MEX")]
+    [InlineData("FSM")]
+    [InlineData("MDA")]
+    [InlineData("MCO")]
+    [InlineData("MNG")]
+    [InlineData("MNE")]
+    [InlineData("MSR")]
+    [InlineData("MAR")]
+    [InlineData("MOZ")]
+    [InlineData("MMR")]
+    [InlineData("NAM")]
+    [InlineData("NRU")]
+    [InlineData("NPL")]
+    [InlineData("NLD")]
+    [InlineData("ANT")]
+    [InlineData("NCL")]
+    [InlineData("NZL")]
+    [InlineData("NIC")]
+    [InlineData("NER")]
+    [InlineData("NGA")]
+    [InlineData("NIU")]
+    [InlineData("NFK")]
+    [InlineData("MNP")]
+    [InlineData("NOR")]
+    [InlineData("OMN")]
+    [InlineData("PAK")]
+    [InlineData("PLW")]
+    [InlineData("PSE")]
+    [InlineData("PAN")]
+    [InlineData("PNG")]
+    [InlineData("PRY")]
+    [InlineData("PER")]
+    [InlineData("PHL")]
+    [InlineData("PCN")]
+    [InlineData("POL")]
+    [InlineData("PRI")]
+    [InlineData("PRT")]
+    [InlineData("QAT")]
+    [InlineData("REU")]
+    [InlineData("ROU")]
+    [InlineData("RUS")]
+    [InlineData("RWA")]
+    [InlineData("BLM")]
+    [InlineData("SHN")]
+    [InlineData("KNA")]
+    [InlineData("LCA")]
+    [InlineData("MAF")]
+    [InlineData("SPM")]
+    [InlineData("VCT")]
+    [InlineData("WSM")]
+    [InlineData("SMR")]
+    [InlineData("STP")]
+    [InlineData("SAU")]
+    [InlineData("SEN")]
+    [InlineData("SRB")]
+    [InlineData("SYC")]
+    [InlineData("SLE")]
+    [InlineData("SGP")]
+    [InlineData("SVK")]
+    [InlineData("SVN")]
+    [InlineData("SLB")]
+    [InlineData("SOM")]
+    [InlineData("ZAF")]
+    [InlineData("SGS")]
+    [InlineData("SSD")]
+    [InlineData("ESP")]
+    [InlineData("LKA")]
+    [InlineData("SDN")]
+    [InlineData("SUR")]
+    [InlineData("SJM")]
+    [InlineData("SWZ")]
+    [InlineData("SWE")]
+    [InlineData("CHE")]
+    [InlineData("SYR")]
+    [InlineData("TWN")]
+    [InlineData("TJK")]
+    [InlineData("TZA")]
+    [InlineData("THA")]
+    [InlineData("TLS")]
+    [InlineData("TGO")]
+    [InlineData("TKL")]
+    [InlineData("TON")]
+    [InlineData("TTO")]
+    [InlineData("TUN")]
+    [InlineData("TUR")]
+    [InlineData("TKM")]
+    [InlineData("TCA")]
+    [InlineData("TUV")]
+    [InlineData("UGA")]
+    [InlineData("UKR")]
+    [InlineData("ARE")]
+    [InlineData("GBR")]
+    [InlineData("USA")]
+    [InlineData("UMI")]
+    [InlineData("URY")]
+    [InlineData("UZB")]
+    [InlineData("VUT")]
+    [InlineData("VEN")]
+    [InlineData("VNM")]
+    [InlineData("VIR")]
+    [InlineData("WLF")]
+    [InlineData("ESH")]
+    [InlineData("YEM")]
+    [InlineData("ZMB")]
+    [InlineData("ZWE")]
+    [Theory]
+    public void Nox_CountryCode3_Constructor_ReturnsSameValue_AllCountries(string countryCode3String)
```

Comment:
Regarding the tests naming, start with Nox I think its redundant and can be removed, does not bring any additional value? What do you think?

CreatingCountryCode3_From_IsValid, does it seems a better name=> Scenario + Test method+ Expected?

This can be applied to all tests in this class?

```suggestion
    public void CreatingCountryCode3_From_IsValid(string countryCode3String)
```